File: KotlinCoffeeShop/Kwon/src/main/java/coffeShop/actor/Shop.java
package coffeShop.actor;

import coffeShop.beverage.*;
import coffeShop.shopdata.Ingredient;
import coffeShop.shopdata.MenuInfo;

import java.util.ArrayList;
import java.util.HashMap;

import static coffeShop.shopdata.Ingredient.coffebean;
import static coffeShop.shopdata.Ingredient.milk;
import static coffeShop.shopdata.Ingredient.water;

public class Shop {



    public ArrayList<HashMap<MenuInfo,Integer>> makeManulist(){
        ArrayList<HashMap<MenuInfo,Integer>> menual = new ArrayList<HashMap<MenuInfo, Integer>>();
        System.out.println("====================================================");

        for(MenuInfo info : MenuInfo.values()){
            HashMap<MenuInfo,Integer> map = new HashMap<MenuInfo,Integer>();
            map.put(info,info.getPrice());
            menual.add(map);
            System.out.println("매뉴명 : " + info.toString() + "가격 : "+ info.getPrice());
        }

        System.out.println("====================================================");
        return menual;
    }

    public Beverage makeBeverage(MenuInfo order){

        Beverage beverage = null;
        String ingredient = null;
        switch (order){
            case esspresso:
                ingredient = coffebean.toString();
                beverage = new Espresso(ingredient,order.toString(),Integer.toString(order.getPrice()));
                break;
            case americano:
                ingredient = coffebean.toString()+" ,"+water.toString();
                beverage = new Americano(ingredient,order.toString(),Integer.toString(order.getPrice()));
                break;
            case capuchino:
                ingredient = coffebean.toString()+" ,"+water.toString()+" ,"+milk.toString();
                beverage = new Capuchino(ingredient,order.toString(),Integer.toString(order.getPrice()));
                break;
            case cafelatte:
                ingredient = coffebean.toString()+" ,"+water.toString()+" ,"+milk.toString();
                beverage = new CafeLatte(ingredient,order.toString(),Integer.toString(order.getPrice()));
                break;
        }
        return beverage;
    }



}

File: KotlinCoffeeShop/Kwon/src/main/kotlin/coffeShop/actor/Shop.kt
package coffeShop.actor

import coffeShop.action.ShopInterface
import coffeShop.props.Beverage
import coffeShop.props.Food
import coffeShop.props.Menu
import coffeShop.shopdata.Ingredient
import coffeShop.shopdata.MenuInfo
import coffeShop.shopdata.MenuType
import coffeShop.shopdata.Recipe

class Shop():ShopInterface {

    var menulist : ArrayList<MenuInfo>? = ArrayList()
    var orderlist : ArrayList<MenuInfo> = ArrayList()
    var productlist : ArrayList<Menu> = ArrayList()
    var price : Int = 0
    var change : Int =0

    override fun sayHello() {
        println("HELLO 뭐드릴까요? ")
    }

    override fun makeMenuList() : ArrayList<MenuInfo> {
        var menutypewriteflag_food : Boolean = false
        var menutypewriteflag_beverage : Boolean = false

        menulist!!.addAll(MenuInfo.values())

        println("--------[MENU]----------")

        for(menu in menulist!!){

            if(menu.type.equals(MenuType.BEVERAGE) && menutypewriteflag_beverage == false ) {
                println("****[Beverage]****")
                menutypewriteflag_beverage = true
            }
            if(menu.type.equals(MenuType.FOOD)  && menutypewriteflag_food == false){
                println("****[Food]****")
                menutypewriteflag_food = true
            }

            println("name : ${menu.name} , price: ${menu.price} , discription: ${menu.discription}")
        }
        println("------------------------")
        return menulist!!
    }

    override fun getOrder(orderlist : ArrayList<MenuInfo>):Int {

        this.orderlist = orderlist
        for(ordermenu in orderlist){
            println("  ${ordermenu.name} ${ordermenu.price} ")
            price+=ordermenu.price
        }
        println("주문합계 ${price}원 입니다.")
        return price
    }

    override fun getMoney(money: Int) :Int {
        if(price < money){
            change = money - price
            println("${money}원 받았습니다. 거스름돈은 ${change}입니다.")
        }
        return change
    }

    override fun makeMenu() : ArrayList<Menu> {

        for(menu in orderlist){
//            이거왜안됨?
//            when(menu.type){
//                is MenuType.FOOD ->....
//            }
            if(menu.type.equals(MenuType.FOOD)){
                var food = Food(menu.price,
                                menu.discription,
                                menu.type.name,
                                Recipe.valueOf(menu.name).getIngredient().toString(),
                                menu.name)

                productlist.add(food)
                println("${food.name}(${food.hashCode()})을 완성했다.")

            }else if(menu.type.equals(MenuType.BEVERAGE)){

                var beverage = Beverage(menu.price,
                                        menu.discription,
                                        menu.type.name,
                                        Recipe.valueOf(menu.name).getIngredient().toString(),
                                        menu.name)

                productlist.add(beverage)
                println("${beverage.name}(${beverage.hashCode()})을 완성했다.")

            }
        }

        return productlist
    }
}
